File: android/src/main/kotlin/dev/steenbakker/flutter_ble_peripheral/models/ServiceDescription.kt
package dev.steenbakker.flutter_ble_peripheral.models

import java.util.UUID

class ServiceDescription(uuid: String, val characteristics: List<CharacteristicDescription>) {

    val uuid = UUID.fromString(uuid)

    constructor(properties: Map<String, Any>) : this(
        properties["uuid"] as String,
        (properties["characteristics"] as List<Map<String,Any>>).map{CharacteristicDescription(it)}
    )
}

File: ios/Classes/Models/ServiceDescription.swift
//
//  ServiceDescription.swift
//  flutter_ble_peripheral
//
//  Created by tiago-bacelar on 26/03/2024.
//

import Foundation
import CoreBluetooth

class ServiceDescription {
    let uuid: CBUUID
    let characteristics: [CharacteristicDescription]
    
    init(properties: Dictionary<String,Any>) {
        uuid = CBUUID(string: properties["uuid"] as! String)
        characteristics = (properties["characteristics"] as! [Dictionary<String,Any>]).map(CharacteristicDescription.init)
    }
}
